components/Elevation/tests/unit/MDCMaterialElevationRespondingTests.m

```diff
@@ -0,0 +1,237 @@
+
+@implementation MDCFakeConformingMDCElevationViewController
+
+- (CGFloat)mdc_currentElevation {
+  return self.elevation;
+}
+
+@end
+
+@interface MDCFakeConformingMDCElevationOverrideView : UIView <MDCElevation>
+@property(nonatomic, assign, readwrite) CGFloat mdc_overrideBaseElevation;
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)(CGFloat elevation);
+@property(nonatomic, assign, readonly) CGFloat mdc_currentElevation;
+@property(nonatomic, assign) CGFloat elevation;
+@end
+
+@implementation MDCFakeConformingMDCElevationOverrideView
+
+- (CGFloat)mdc_currentElevation {
+  return self.elevation;
+}
+
+@end
+
+@interface MDCFakeConformingMDCElevationOverrideViewController : UIViewController <MDCElevation>
+@property(nonatomic, assign, readwrite) CGFloat mdc_overrideBaseElevation;
+@property(nonatomic, copy, nullable) void (^mdc_elevationDidChangeBlock)(CGFloat elevation);
+@property(nonatomic, assign, readonly) CGFloat mdc_currentElevation;
+@property(nonatomic, assign) CGFloat elevation;
+@end
+
+@implementation MDCFakeConformingMDCElevationOverrideViewController
+
+- (CGFloat)mdc_currentElevation {
+  return self.elevation;
+}
+
+@end
+
+@interface MDCMaterialElevationRespondingTests : XCTestCase
+@property(nonatomic, strong, nullable) UIView *fakeView;
+@property(nonatomic, strong, nullable) MDCFakeConformingMDCElevationView *fakeElevationView;
+@property(nonatomic, strong, nullable)
+    MDCFakeConformingMDCElevationOverrideView *fakeElevationOverrideView;
+@property(nonatomic, strong, nullable) UIViewController *fakeViewController;
+@property(nonatomic, strong, nullable)
+    MDCFakeConformingMDCElevationViewController *fakeElevationViewController;
+@property(nonatomic, strong, nullable)
+    MDCFakeConformingMDCElevationOverrideViewController *fakeElevationOverrideViewController;
+@end
+
+@implementation MDCMaterialElevationRespondingTests
+
+- (void)setUp {
+  [super setUp];
+
+  self.fakeView = [[UIView alloc] init];
+  self.fakeElevationView = [[MDCFakeConformingMDCElevationView alloc] init];
+  self.fakeElevationOverrideView = [[MDCFakeConformingMDCElevationOverrideView alloc] init];
+  self.fakeViewController = [[UIViewController alloc] init];
+  self.fakeElevationViewController = [[MDCFakeConformingMDCElevationViewController alloc] init];
+  self.fakeElevationOverrideViewController =
+      [[MDCFakeConformingMDCElevationOverrideViewController alloc] init];
+}
+
+- (void)tearDown {
+  self.fakeElevationOverrideViewController = nil;
+  self.fakeElevationViewController = nil;
+  self.fakeViewController = nil;
+  self.fakeElevationOverrideView = nil;
+  self.fakeElevationView = nil;
+  self.fakeView = nil;
+
+  [super tearDown];
+}
+
+- (void)testViewInElevationView {
+  // Given
+  CGFloat fakeElevation = 3;
+  self.fakeElevationView.elevation = fakeElevation;
+
+  // When
+  [self.fakeElevationView addSubview:self.fakeView];
+
+  // Then
+  XCTAssertEqual(self.fakeView.mdc_baseElevation, fakeElevation);
```

Comment:
Nit: Please use `XCTAssertEqualWithAccuracy` for floating-point comparisons here and throughout.

Reply:
Done